caom2/caom2/shape.py

```diff
@@ -90,20 +91,230 @@ class SegmentType(Enum):
 
-    def __init__(self):
-        pass
+    def __init__(self, center,
+                 width, height):
+        """
+        Initialize a Box instance
+        """
+        self.center = center
+        self.width = width
+        self.height = height
+
+    def get_area(self):
+        """ TODO: this is cartesian approximation, use spherical geom? """
+        return self._width * self._height
+
+    def get_size(self):
+        return math.sqrt(self._width * self._width + self._height * self._height)
+
+    # Properties
+
+    @property
+    def center(self):
+        """
+        type: Point
+        """
+        return self._center
+
+    @center.setter
+    def center(self, value):
+        caom_util.type_check(value, Point, 'center', override=False)
+        self._center = value
+
+    @property
+    def width(self):
+        """
+        type: float
+        """
+        return self._width
+
+    @width.setter
+    def width(self, value):
+        assert value > 0, "width must be positive"
+        caom_util.type_check(value, float, 'width', override=False)
+        self._width = value
+
+    @property
+    def height(self):
+        """
+        type: float
+        """
+        return self._height
+
+    @height.setter
+    def height(self, value):
+        assert value > 0, "height must be positive"
+        caom_util.type_check(value, float, 'height', override=False)
+        self._height = value
 
 
 class Circle(common.CaomObject):
 
-    def __init__(self):
-        pass
+    def __init__(self, center,
+                 radius):
+        """
+        Initialize a Circle instance
+        """
+        self.center = center
+        self.radius = radius
+
+    def get_area(self):
+        """ TODO: this is cartesian approximation, use spherical geom? """
+        return math.pi * self._radius * self._radius
+
+    def get_size(self):
+        return 2.0 * self._radius
+
+    # Properties
+
+    @property
+    def center(self):
+        """
+        type: Point
+        """
+        return self._center
+
+    @center.setter
+    def center(self, value):
+        caom_util.type_check(value, Point, 'center', override=False)
+        self._center = value
+
+    @property
+    def radius(self):
+        """
+        type: float
+        """
+        return self._radius
+
+    @radius.setter
+    def radius(self, value):
+        assert value > 0, "radius must be positive"
+        caom_util.type_check(value, float, 'radius', override=False)
+        self._radius = value
+
+
+class SubInterval(common.CaomObject):
+
+    CTOR_UTYPES = ["lower", "upper"]
+
+    def __init__(self, lower, upper):
+        self.lower = lower
+        self.upper = upper
+
+    # Properties
+
+    @property
+    def lower(self):
+        """
+        type: float
+        """
+        return self._lower
+
+    @lower.setter
+    def lower(self, value):
+        caom_util.type_check(value, float, 'lower', override=False)
+        has_upper = True
+        try:
```

Comment:
Not sure what 'this' refers to. If it refers to the type_check(), then we want to ensure that the value is of type float. If it refers to the usage of has_upper, we want to determine if the local variable 'upper' has been set. If it has, we want to ensure that we do not have upper < lower.